nomad/state/sorting.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package state
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-memdb"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+// SortOption represents how results can be sorted.
+type SortOption bool
+
+const (
+	// SortDefault indicates that the result should be returned using the
+	// default go-memdb ResultIterator order.
+	SortDefault SortOption = false
+
+	// SortReverse indicates that the result should be returned using the
+	// reversed go-memdb ResultIterator order.
+	SortReverse SortOption = true
+)
+
+// QueryOptionSort returns the appropriate SortOption for given QueryOptions.
+func QueryOptionSort(qo structs.QueryOptions) SortOption {
+	return SortOption(qo.Reverse)
+}
+
+// GetSorted executes either txn.Get() or txn.GetReverse()
+// depending on the provided SortOption.
+func GetSorted(txn *txn, sort SortOption, table, index string, args ...any) (memdb.ResultIterator, error) {
```

Comment:
this little func is the main helper that I like, the rest grew out of this.

Reply:
Should this really be exported? This exposes internal helper functions of the state store to the caller. But more importantly, the `txn` type isn't exported, so it's not like you can call this outside of the `nomad/state` package anyways.